dispatch/apps/api/views.py

```diff
@@ -407,6 +407,42 @@ def articles(self, request, pk=None):
         view = ArticleViewSet.as_view({'get': 'topic'})
         return view(request, pk=pk)
 
+class FileViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for File model views.
+    """
+    model = File
+    serializer_class = FileSerializer
+
+    def get_queryset(self):
+        queryset = File.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            # If a search term (q) is present, filter queryset by term against `name`
+            queryset = queryset.filter(name__icontains=q)
+        return queryset
+
+    def bulk_delete(self, request):
+        deleted = []
+        ids = self.request.data.get('ids', None)
+
+        if ids is not None:
+            ids = ids.split(',')
```

Comment:
Also, why the need to split ids? Why is the frontend passing a comma-separated string? Would it not be far more sane to pass a json array of ints?